openedx/openedx-platform
# Registration form: a "Retype password" field and verification that a email really exists

**Background:** Users often misspell there emails and passwords and can't login a 2d time

**Studio Updates:** None.

**LMS Updates:** Regitration form validation changed: a "Retype password" must be entered and be equal to "Password". A email is verified with MX record and communication to SMTP server. Both options are optional and are configured in lms.env.json.

**Manual test instructions:**
Change lms.env.json:
`"REGISTRATION_EMAIL_FULL_VERIFICATION":true`
`"REGISTRATION_EXTRA_FIELDS": {
        "password_copy": "required", 
`
    Open http://127.0.0.1:8000/register and try to:
    1. enter different "Password" and "Retype password" and see a validation message;
    2. enter a email with invalid domain and see a validation message;
    3. enter a not existing email with valid domain and see a validation message (this may not work for some domains);
![edx-105-1](https://cloud.githubusercontent.com/assets/7777978/14743858/85e37e4e-08ac-11e6-97b7-37168c4d274c.jpg)
![edx-105-2](https://cloud.githubusercontent.com/assets/7777978/14743862/8eee3d94-08ac-11e6-9425-db9252dc6a27.jpg)

The pull request includes a new dependency on dnspython.
